constructor/header.sh

```diff
@@ -597,8 +601,7 @@ if [ "$TEST" = "1" ]; then
      fi
      conda index "$PREFIX"/conda-bld/__PLAT__/
      conda-build --override-channels --channel local --test --keep-going "$PREFIX"/conda-bld/__PLAT__/*.tar.bz2
-    )
-    NFAILS=$?
+    ) || NFAILS=$?
```

Comment:
Why is this change needed? Shouldn't the other one behaev in the same way?

Reply:
With `set -e` being set, the shell would stop with an error before it evaluates `NFAILS=$?` when it is in a *newline*.

If the subshell expression is extended in the same line via `$( code ) || extra_code`, then the extra_code is executed in case of error allowing error code evaluation.

Reply:
Love it, thanks!